sources/models/src/modeled_types/shared.rs

```diff
@@ -250,3 +252,99 @@ mod test_url {
         }
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// FriendlyVersion represents a valid version string that can be prefixed with or without 'v.
+/// It can also be set to 'latest' to represent the latest version. It stores the original string
+/// and makes it accessible through standard traits.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct FriendlyVersion {
+    inner: String,
+}
+
+impl TryFrom<&str> for FriendlyVersion {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        if input == "latest" {
+            return Ok(FriendlyVersion {
+                inner: input.to_string(),
+            });
+        }
+        // If the string begins with a 'v', skip it before checking if it is valid semver.
+        let version = if input.starts_with('v') {
+            &input[1..]
+        } else {
+            input
+        };
+
+        if version.parse::<semver::Version>().is_ok() {
+            return Ok(FriendlyVersion {
+                inner: input.to_string(),
+            });
+        }
+        error::InvalidVersion { input }.fail()
+    }
+}
+
+impl TryInto<semver::Version> for FriendlyVersion {
```

Comment:
I think you can reverse this and do `TryFrom<FriendlyVersion> for semver::Version` to be more consistent with the other implementations.